lib/capify-ec2.rb

```diff
@@ -372,13 +425,55 @@ def deregister_instance_from_elb_by_dns(server_dns)
   end
 
   def reregister_instance_with_target_group_by_dns(server_dns, target_group, timeout)
-    # Stubbed method
-    # Return `reregistered`
+    instance = get_instance_by_dns(server_dns)
+
+    # Check if the instance is a VPC instance, if not return
+    if not is_vpc_instance?(instance)
+        puts "[Capify-EC2] Skipping non-VPC instance '#{server_dns}' ('#{instance}') from target group '#{target_group}'..."
+        return []
+    end
+
+    # Sleep, but why?
+    sleep 10
+    
+    puts "[Capify-EC2] Re-registering instance with ALB target group '#{target_group}'..."
+
+    # Instance registration requires the ALB target group ARN and the instance ID
+    # Obtain target group ARN from target group name assuming the name is unique.
+    target_group_arn = alb_client.describe_target_groups({
+        names: [target_group],
+    })['TargetGroups'][0]['TargetGroupArn']
+
+    # Register the instance with the target group ARN
+    client.register_targets({
+        target_group_arn: target_group_arn,
+        targets: [ { id: instance } ]
+    })
+
+    # Verify instance is attached
+
+    # During registration we would expect to see a state of
+    # initial or healthy.
+    if %w(initial healthy).include?(response.target_health_descriptions[0].target_health.state)
```

Comment:
Can I just confirm this - in the code below for ELB it does a timeout, and so my understanding here is that the target group functions differently and immediately goes to initial or healthy?

What would happen if you registered an instance with a failing health check? Would it go immediately to failed (skipping initial)?

Reply:
It would always go to initial to begin with until it's gone through a period of health checks. We can loop here until the instance goes healthy at the expense of longer deployment times. Happy to add that in now if you want?

Reply:
Health checks added.